src/PrestaShopBundle/Form/Admin/Sell/Product/CreateProductFormType.php

```diff
@@ -0,0 +1,104 @@
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+declare(strict_types=1);
+
+namespace PrestaShopBundle\Form\Admin\Sell\Product;
+
+use PrestaShop\PrestaShop\Core\Domain\Product\ValueObject\ProductType;
+use PrestaShop\PrestaShop\Core\Form\FormChoiceAttributeProviderInterface;
+use PrestaShop\PrestaShop\Core\Form\FormChoiceProviderInterface;
+use PrestaShopBundle\Form\Admin\Type\TranslatorAwareType;
+use Symfony\Component\Form\Extension\Core\Type\ChoiceType;
+use Symfony\Component\Form\Extension\Core\Type\SubmitType;
+use Symfony\Component\Form\FormBuilderInterface;
+use Symfony\Component\OptionsResolver\OptionsResolver;
+use Symfony\Component\Translation\TranslatorInterface;
+
+class CreateProductFormType extends TranslatorAwareType
+{
+    /**
+     * @var FormChoiceProviderInterface|FormChoiceAttributeProviderInterface
+     */
+    private $formChoiceProvider;
+
+    /**
+     * @param TranslatorInterface $translator
+     * @param array $locales
+     * @param FormChoiceProviderInterface|FormChoiceAttributeProviderInterface $formChoiceProvider
+     */
+    public function __construct(
+        TranslatorInterface $translator,
+        array $locales,
+        $formChoiceProvider
+    ) {
+        parent::__construct($translator, $locales);
+        $this->formChoiceProvider = $formChoiceProvider;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    public function buildForm(FormBuilderInterface $builder, array $options)
+    {
+        $builder
+            ->add('type', ChoiceType::class, [
+                'choices' => $this->formChoiceProvider->getChoices(),
+                'choice_attr' => $this->formChoiceProvider->getChoicesAttributes(),
+                'required' => true,
+                'label' => false,
+                'empty_data' => ProductType::TYPE_STANDARD,
+                'block_prefix' => 'product_type',
+            ])
+            ->add('create', SubmitType::class, [
+                'label' => $this->trans('Add new product', 'Admin.Catalog.Help'),
+                'row_attr' => [
+                    'class' => 'text-center',
+                ],
+            ])
+        ;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    public function configureOptions(OptionsResolver $resolver)
+    {
+        $resolver->setDefaults([
+            'required' => false,
+            'label' => $this->trans('Add new product', 'Admin.Catalog.Help'),
+            'label_tag_name' => 'h3',
+        ]);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    public function getBlockPrefix()
```

Comment:
You can type-hint return of this method with `string`

Reply:
Same answer as here https://github.com/PrestaShop/PrestaShop/pull/26024/files#r746537203